File: examples/backends/fastapi/app/routes/me.py
"""GET /auth/me — Return user info from ID token claims."""

from fastapi import APIRouter, Depends, Request
from fastapi.responses import JSONResponse

from ..cognito import decode_jwt_payload
from ..dependencies import require_auth

router = APIRouter()


@router.get("/auth/me")
async def get_me(request: Request, tokens: dict = Depends(require_auth)):
    try:
        claims = decode_jwt_payload(tokens["id_token"])
        return {
            "email": claims.get("email"),
            "sub": claims.get("sub"),
            "groups": claims.get("cognito:groups", []),
        }
    except Exception:
        return JSONResponse({"error": "Failed to decode token"}, status_code=500)

File: rust/src/routes/me.rs
//! GET /auth/me

use axum::Json;

use crate::cognito::jwt::decode_jwt_unverified;
use crate::error::AppError;
use crate::ocsf;
use crate::session::middleware::SessionHandle;
use crate::types::{SessionTokens, UserInfoResponse};

/// Return user info decoded from the session's ID token (unverified).
pub async fn me(session: SessionHandle) -> Result<Json<UserInfoResponse>, AppError> {
    let tokens: SessionTokens = session.tokens().await?;

    let claims = decode_jwt_unverified(&tokens.id_token).map_err(|e| {
        ocsf::authentication_event(
            ocsf::ACTIVITY_OTHER,
            "Other",
            ocsf::STATUS_FAILURE,
            ocsf::SEVERITY_MEDIUM,
            None,
            ocsf::AUTH_PROTOCOL_OAUTH2,
            "OAuth 2.0/OIDC",
            &format!("User info retrieval failed: {e}"),
        );
        AppError::TokenDecodeFailed
    })?;

    Ok(Json(UserInfoResponse {
        email: claims.email,
        sub: Some(claims.sub),
        groups: claims.cognito_groups,
    }))
}
